ouroboros-consensus-cardano/src/unstable-cardano-tools/Cardano/Tools/ImmDBServer/Json/SendRecv.hs

```diff
@@ -15,19 +14,19 @@ data Direction = Recv | Send
 
 -- | The event as it appears in the log file.
 data SendRecvEvent muxtm tm cnt = MkSendRecvEvent {
-        at :: tm
+        at           :: tm
         -- ^ when
-      , mux_at :: muxtm
+      , mux_at       :: Maybe muxtm
         -- ^ For 'Recv', when these bytes left the demuxer.
         --
         -- For 'Send', when the mini protocol began trying to put these bytes
         -- into the muxer.
-      , direction :: Direction
-      , msg :: String
+      , direction    :: Direction
+      , msg          :: Aeson.Value
```

Comment:
Not sure why this isn't a type var `msg` with `Aeson.(To|From)Object msg` instances? But not really important at this point.